### ISSUE TITLE
"Last Updated" not updating in real-time-html report

### ISSUE THREAD

**Issue by GainfulShrimp** [allinurl/goaccess#1548]
I've had the real-time-html report working well with nginx reverse proxy for some time now (using v1.2).

After moving to v1.3, the "Last Updated" value (top-right of the report) doesn't update whenever the server pushes a websocket update, as it did before.

I'm not sure whether this is intentional or a bug, but I miss the feature so would be grateful for any help on how to get the updating date/time to work again please.

FYI, everything looks looks fine with the wss: feed in Chrome's dev tools: when I reload a page on my blog, I can see a websocket message is pushed by goaccess to the frontend almost immediately, and the figures in the report update just fine.  I would like the 'Last Updated' value to be updated at the same time, as it was in v1.2, if possible...

**Comment by allinurl** [comment]
Yes, this is a regression which I think was fixed upstream. Can I ask you to please try building from [development](https://goaccess.io/download#build) and let me know if that addresses the issue. Thanks.

**Comment by GainfulShrimp** [comment]
Thanks for getting back to me.  

Unfortunately, the latest development version made things worse: the 'Last Updated' still didn't update, but also most of the panels on the bottom half of the report - i.e. below 'unique visitors' and 'requested files' - would not populate until I clicked one of the pagination (< or >) buttons.

I've gone back to 1.3 release for now.

Happy to try the dev build again if/when you make further changes, of course.

**Comment by allinurl** [comment]
Thanks for the update. For some reason I thought this was already addressed as I do remember this issue. Let me look into it and I'll post back.

**Comment by skyrim61** [comment]
I meet the same problem as above

**Comment by GainfulShrimp** [comment]
Any updates with this issue please?

**Comment by pomusa** [comment]
if you are using centos... couple of thing to check if you are on Centos. 
1. if you have SeLinux running, turn it off 
# getenforce  -- to check
if you have permissive Enforced, turn it off
e.g.
# getenforce
Enforcing
# setenforce Permissive

2. Check your firewall, if you have  the port 7890 enabled
# firewall-cmd --list-ports
# firewall-cmd --list-services

e.g.
# firewall-cmd --list-services
cockpit dhcpv6-client http https ssh
# firewall-cmd --list-ports
7890/tcp

if you don't, then you need to add
# firewall-cmd --permanent --add-port=7890/tcp
# firewall-cmd --reload

I just installed the go-access-1.3 and it works for me on Centos 8 running on apache 2.4.35. I tested it on Non SSL first .. to make sure it works before trying to use SSL.

**Comment by GainfulShrimp** [comment]
Thanks for the comment @pomusa, but I don't think it's relevant to this issue, unfortunately.

I don't use Centos anyway, but the problem is that I've got websocket connections working fine, yet the 'Last Updated' value never changes (the statistics/charts do though).

**Comment by allinurl** [comment]
@GainfulShrimp Working on this...

**Comment by allinurl** [comment]
This has been fixed upstream. Please feel free to build from development and let me know how it goes. Else, will be deployed in the upcoming release.

**Comment by swelanauguste** [comment]
thanks for all your help @allinurl. I am still having this issue but whenever I restart the server it will update that stats.

**Comment by allinurl** [comment]
@swelanauguste Are you currently utilizing the most recent version? Additionally, do you observe the green dot adjacent to the settings, as demonstrated on the https://rt.goaccess.io/ sample page?

**Comment by swelanauguste** [comment]
@allinurl in docker-compose i am using:

`version: '3.3'
services:
    goaccess:
        image: 'xavierh/goaccess-for-nginxproxymanager:latest'
        container_name: goaccess
        restart: always
        ports:
            - '7880:7880'
        environment:
            - PUID=1000
            - PGID=1000
            - TZ=America/St_Lucia
            - SKIP_ARCHIVED_LOGS=False #optional
            - DEBUG=True #optional
            - BASIC_AUTH=False #optional
            - BASIC_AUTH_USERNAME=user #optional
            - BASIC_AUTH_PASSWORD=pass #optional
            - EXCLUDE_IPS=127.0.0.1,10.10.10.100 #optional - comma delimited
            - LOG_TYPE=NPM #optional - more information below
        volumes:
            - ~/kingship-nginx-proxy-manager/data/logs:/opt/log
        networks:
          - kingship-nginx-proxy-manager_default
networks:
  kingship-nginx-proxy-manager_default:
    external: true`

I do not have the green dot in my personal setup, and neither do I see one in your link.

mine is at https://goaccess.kingship.info/

**Comment by allinurl** [comment]
@swelanauguste It appears that you're encountering problems with your Docker setup and ports. Could you please provide the output from your web browser's console? Additionally, could you share the logs from your Docker instance?

**Comment by swelanauguste** [comment]
=================Docker logs=================

RUN NPM GOACCESS
 [SETTING UP STORAGE /goaccess-logs/archives/proxy-host-12_access.log.1] {0} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1814} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-12_access.log.2] {353} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {205} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.1] {240} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.1] {926} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.2] {1694} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-8_access.log.1] {1764} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-8_access.log.1] {4455} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {1883} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {4714} @ {4714/s}
 [PARSING /goaccess-logs/archives/proxy-host-7_access.log.1] {957} @ {957/s}
 [PARSING /goaccess-logs/archives/proxy-host-7_access.log.1] {3872} @ {3872/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.4] {108} @ {108/s}

WebSocket server ready to accept new client connections

GOAN v1.1.19
NGINX SETUP...
NGINX BASIC AUTH
-------------------------------
None
NPM INSTANCE SETTING UP...
LOADING NPM PROXY LOGS
-------------------------------
	Adding proxy logs...
	File /opt/log/proxy-host-7_access.log exists
	File /opt/log/proxy-host-8_access.log exists
	File /opt/log/proxy-host-10_access.log exists
	File /opt/log/proxy-host-1_access.log exists
	File /opt/log/proxy-host-13_access.log exists
	File /opt/log/proxy-host-3_access.log exists
	File /opt/log/proxy-host-5_access.log exists
	File /opt/log/proxy-host-11_access.log exists
	File /opt/log/proxy-host-12_access.log exists
	File /opt/log/proxy-host-9_access.log exists
	File /opt/log/proxy-host-2_access.log exists
	File /opt/log/proxy-host-4_access.log exists
	File /opt/log/proxy-host-6_access.log exists
	Found (13) proxy logs...
	SKIP ARCHIVED LOGS
	-------------------------------
	FALSE
	Adding proxy archive logs...
	File /opt/log/proxy-host-3_access.log.1.gz exists
	File /opt/log/proxy-host-9_access.log.2.gz exists
	File /opt/log/proxy-host-7_access.log.3.gz exists
	File /opt/log/proxy-host-12_access.log.2.gz exists
	File /opt/log/proxy-host-4_access.log.2.gz exists
	File /opt/log/proxy-host-8_access.log.4.gz exists
	File /opt/log/proxy-host-4_access.log.4.gz exists
	File /opt/log/proxy-host-6_access.log.3.gz exists
	File /opt/log/proxy-host-3_access.log.4.gz exists
	File /opt/log/proxy-host-7_access.log.2.gz exists
	File /opt/log/proxy-host-6_access.log.2.gz exists
	File /opt/log/proxy-host-2_access.log.2.gz exists
	File /opt/log/proxy-host-5_access.log.3.gz exists
	File /opt/log/proxy-host-4_access.log.3.gz exists
	File /opt/log/proxy-host-9_access.log.3.gz exists
	File /opt/log/proxy-host-2_access.log.4.gz exists
	File /opt/log/proxy-host-6_access.log.4.gz exists
	File /opt/log/proxy-host-1_access.log.3.gz exists
	File /opt/log/proxy-host-9_access.log.1.gz exists
	File /opt/log/proxy-host-7_access.log.4.gz exists
	File /opt/log/proxy-host-5_access.log.2.gz exists
	File /opt/log/proxy-host-3_access.log.3.gz exists
	File /opt/log/proxy-host-9_access.log.4.gz exists
	File /opt/log/proxy-host-4_access.log.1.gz exists
	File /opt/log/proxy-host-3_access.log.2.gz exists
	File /opt/log/proxy-host-2_access.log.1.gz exists
	File /opt/log/proxy-host-2_access.log.3.gz exists
	File /opt/log/proxy-host-5_access.log.4.gz exists
	File /opt/log/proxy-host-11_access.log.1.gz exists
	File /opt/log/proxy-host-8_access.log.1.gz exists
	File /opt/log/proxy-host-11_access.log.2.gz exists
	File /opt/log/proxy-host-8_access.log.2.gz exists
	File /opt/log/proxy-host-8_access.log.3.gz exists
	File /opt/log/proxy-host-7_access.log.1.gz exists
	File /opt/log/proxy-host-1_access.log.1.gz exists
	File /opt/log/proxy-host-6_access.log.1.gz exists
	File /opt/log/proxy-host-1_access.log.2.gz exists
	File /opt/log/proxy-host-1_access.log.4.gz exists
	File /opt/log/proxy-host-5_access.log.1.gz exists
	File /opt/log/proxy-host-13_access.log.1.gz exists
	File /opt/log/proxy-host-12_access.log.1.gz exists
	Added (41) proxy archived logs from /opt/log...
EXCLUDE IPS
-------------------------------

1

10.10.10.100
Setting GeoIP Database
-------------------------------
DEFAULT
DEBUG
-------------------------------
ON
RUN NPM GOACCESS
 [SETTING UP STORAGE /goaccess-logs/archives/proxy-host-12_access.log.1] {0} @ {0/s}
 [PARSING /opt/log/proxy-host-7_access.log] {1} @ {0/s}
 [PARSING /opt/log/proxy-host-7_access.log] {1} @ {0/s}
 [PARSING /opt/log/proxy-host-7_access.log] {1} @ {0/s}
 [PARSING /opt/log/proxy-host-7_access.log] {1} @ {0/s}
 [PARSING /opt/log/proxy-host-7_access.log] {1} @ {0/s}
 [PARSING /opt/log/proxy-host-2_access.log] {0} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {277} @ {277/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {277} @ {277/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {277} @ {277/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {615} @ {615/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1029} @ {1029/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1030} @ {1030/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1034} @ {1034/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1034} @ {1034/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1154} @ {1154/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1155} @ {577/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {1984} @ {992/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {2315} @ {1157/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.1] {2315} @ {1157/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {10} @ {5/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {10} @ {5/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {10} @ {5/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {10} @ {5/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {736} @ {368/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {742} @ {371/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {744} @ {248/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {750} @ {250/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {755} @ {251/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {772} @ {257/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.2] {780} @ {260/s}
 [PARSING /goaccess-logs/archives/proxy-host-7_access.log.3] {10} @ {3/s}
 [PARSING /goaccess-logs/archives/proxy-host-12_access.log.2] {82} @ {27/s}
 [PARSING /goaccess-logs/archives/proxy-host-12_access.log.2] {604} @ {201/s}
 [PARSING /goaccess-logs/archives/proxy-host-12_access.log.2] {655} @ {218/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.2] {27} @ {9/s}
 [PARSING /goaccess-logs/archives/proxy-host-8_access.log.4] {64} @ {16/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {52} @ {13/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {52} @ {13/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {52} @ {13/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {59} @ {14/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {97} @ {24/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {412} @ {103/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.4] {833} @ {208/s}
 [PARSING /goaccess-logs/archives/proxy-host-7_access.log.2] {117} @ {29/s}
 [PARSING /goaccess-logs/archives/proxy-host-5_access.log.3] {85} @ {21/s}
 [PARSING /goaccess-logs/archives/proxy-host-9_access.log.3] {5} @ {1/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.3] {18} @ {3/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.3] {65} @ {13/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.3] {106} @ {21/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.3] {160} @ {32/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.3] {221} @ {44/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.3] {282} @ {56/s}
 [PARSING /goaccess-logs/archives/proxy-host-5_access.log.2] {10} @ {2/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.3] {46} @ {9/s}
 [PARSING /goaccess-logs/archives/proxy-host-4_access.log.1] {165} @ {33/s}
 [PARSING /goaccess-logs/archives/proxy-host-3_access.log.2] {675} @ {112/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.1] {1} @ {0/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.1] {44} @ {7/s}
 [PARSING /goaccess-logs/archives/proxy-host-8_access.log.1] {2319} @ {386/s}
 [PARSING /goaccess-logs/archives/proxy-host-8_access.log.1] {4992} @ {832/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {90} @ {15/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {182} @ {30/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {255} @ {42/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {335} @ {55/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {337} @ {56/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {404} @ {57/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {1859} @ {265/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {4629} @ {661/s}
 [PARSING /goaccess-logs/archives/proxy-host-11_access.log.2] {5501} @ {785/s}
 [PARSING /goaccess-logs/archives/proxy-host-7_access.log.1] {590} @ {84/s}
 [PARSING /goaccess-logs/archives/proxy-host-7_access.log.1] {3488} @ {498/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.1] {52} @ {7/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.1] {166} @ {23/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.2] {176} @ {25/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.2] {549} @ {78/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.2] {674} @ {84/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.2] {894} @ {111/s}
 [PARSING /goaccess-logs/archives/proxy-host-1_access.log.4] {16} @ {2/s}
 [PARSING /goaccess-logs/archives/proxy-host-5_access.log.1] {45} @ {5/s}
 [PARSING /goaccess-logs/archives/proxy-host-13_access.log.1] {839} @ {104/s}

WebSocket server ready to accept new client connections
=================Docker logs=================

============Web Browser's Console=============
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
DevTools failed to load source map: Could not load content for chrome-extension://liecbddmkiiihnedobmlmillhodjkdmb/js/companionBubble.js.map: System error: net::ERR_BLOCKED_BY_CLIENT
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
setWebSocket @ (index):174
reconnect @ (index):154
(anonymous) @ (index):196
(index):174 WebSocket connection to 'wss://goaccess.kingship.info/' failed: 
============Web Browser's Console=============

Thanks @allinurl and I hope these help

**Comment by swelanauguste** [comment]
@allinurl I just tried my goaccess page with only the IP:PORT (http://192.168.0.100:7880/) and I am getting the green dot next to the gear icon and an real-time update I was using with Nginx Proxy Manager. I enabled web sockets and it works now. Thanks again, your questions pointed me in the right direction.

<img width="496" alt="Screenshot 2023-09-18 at 6 56 06 PM" src="https://github.com/allinurl/goaccess/assets/51264116/2cd5cdc6-7a8f-4ad3-81f2-97dae5adf64a">
